### PR TITLE
Adds armor support for ties

### PR DESCRIPTION
**Pull request by Core0verload**
Armor values on ties are supported now.

Hunter's talisman now actually adds
`armor = list(melee = 5, bullet = 5, laser = 5, energy = 5, bomb = 20, bio = 20, rad = 5)`
to jumpsuit its attached to.

This PR also adds support for ties having pockets, but, unlike #19678, doesn't add any such item.

### CODE REVIEW

File: code/modules/clothing/under/ties.dm [hunk 1]
```diff
@@ -312,7 +355,6 @@
 	desc = "Damn, it feels good to be a gangster."
```
Comment (GunHog) [inline code/modules/clothing/under/ties.dm]: @PKPenguin321, did you at one point want to put armor on this item?

**Comment by Fox-McCloud** [comment]
random fact: Bio armor is pointless

File: code/modules/clothing/under/ties.dm (hunk 1 above)
Comment (PKPenguin321) [inline code/modules/clothing/under/ties.dm]: Yeah, these things were originally like mini ablatives

Once this goes through I could probably make a gang variant with minor armor or whatever

**Comment by TechnoAlchemisto** [comment]
Doesn't bio armor limit reagent transfer?

File: code/modules/clothing/under/ties.dm [hunk 2]
```diff
@@ -322,9 +364,6 @@
 /obj/item/clothing/tie/talisman
 	name = "bone talisman"
 	desc = "A hunter's talisman, some say the old gods smile on those who wear it."
-	icon = 'icons/obj/clothing/ties.dmi'
 	icon_state = "talisman"
-	item_state = ""
 	item_color = "talisman"
-	body_parts_covered = CHEST|GROIN|LEGS|FEET|ARMS|HANDS|HEAD
-	armor = list(melee = 5, bullet = 5, laser = 5, energy = 5, bomb = 50, bio = 65, rad = 5) //Faith is the best armor. //This won't actually work because of accessories kill me with a fucking knife jesus christ I hate code
+	armor = list(melee = 5, bullet = 5, laser = 5, energy = 5, bomb = 20, bio = 20, rad = 5) //Faith is the best armor.
```
Comment (TechnoAlchemisto) [inline code/modules/clothing/under/ties.dm]: Does removing body parts covered mean armor applies to all areas?

Comment (Core0verload) [inline code/modules/clothing/under/ties.dm]: Ties apply armor to all areas covered by jumpsuit.

**Comment by Fox-McCloud** [comment]
@TechnoAlchemisto 

Nope. All that jazz runs off of permeability coefficients.

File: code/modules/clothing/under/ties.dm (hunk 2 above)
Comment (TechnoAlchemisto) [inline code/modules/clothing/under/ties.dm]: Would keeping the body_parts_covered tag apply it to the head? Or does it get overrided to just cover what the jumpsuit does?

Comment (Fox-McCloud) [inline code/modules/clothing/under/ties.dm]: keeping the body parts covered would do nothing.

Comment (TechnoAlchemisto) [inline code/modules/clothing/under/ties.dm]: fuckin' rip, I wanted that talisman to give 5 armor for each region

Comment (Fox-McCloud) [inline code/modules/clothing/under/ties.dm]: Technically speaking, he could add code to make the talisman make the jumpsuit cover all body parts if he really wanted.

File: code/game/objects/items/weapons/storage/internal.dm [hunk 3]
```diff
@@ -35,6 +35,9 @@
 /obj/item/weapon/storage/internal/pocket/big
 	max_w_class = 3
 
+/obj/item/weapon/storage/internal/pocket/wide
```
Comment (optimumtact) [inline code/game/objects/items/weapons/storage/internal.dm]: please take this out of your PR

File: code/game/objects/items/weapons/storage/internal.dm [hunk 4]
```diff
@@ -44,6 +47,13 @@
 	max_w_class = 1
 	priority = FALSE
 
+/obj/item/weapon/storage/internal/pocket/holster
```
Comment (optimumtact) [inline code/game/objects/items/weapons/storage/internal.dm]: Please take this out as well.

**Comment by Core0verload** [comment]
@optimumtact done

**Comment by optimumtact** [comment]
Cheers

**Pull request merged**